warehouse/query-core/src/main/java/datawave/query/jexl/JexlNodeBuilder.java

```
@@ -0,0 +1,637 @@
+     * 
+     * @param children
+     *            the children nodes
+     * @return this builder
+     */
+    public JexlNodeBuilder<T> withChildren(JexlNode... children) {
+        for (JexlNode child : children) {
+            withChild(child);
+        }
+        return this;
+    }
+    
+    /**
+     * Add each node in the provided iterable as a child to this builder.
+     * 
+     * @param children
+     *            the children nodes
+     * @return this builder
+     */
+    public JexlNodeBuilder<T> withChildren(Iterable<? extends JexlNode> children) {
+        children.forEach(this::withChild);
+        return this;
+    }
+    
+    /**
+     * Add a child node builder to this builder. The final child will be built when {@link JexlNodeBuilder#build()} is called on this builder.
+     * 
+     * @param builder
+     *            the builder
+     * @return this builder
+     */
+    public JexlNodeBuilder<T> withChild(JexlNodeBuilder<?> builder) {
+        children.add(builder::build);
+        return this;
+    }
+    
+    /**
+     * Build and return the new {@link JexlNode} instance with the specified image, value, and children. Any child nodes or child node suppliers that were
+     * provided to this builder that are either null or result in null nodes will not be included in the node's children.
+     * 
+     * @return the node
+     */
+    public T build() {
+        // Create the initial node.
+        T node = constructor.get();
+        node.image = this.image;
+        node.jjtSetValue(this.value);
+        
+        // Create the finalized children.
+        // @formatter:off
+        JexlNode[] childArray = children.stream()
+                        .filter(Objects::nonNull) // Filter out any possible null suppliers.
+                        .map(Supplier::get)       // Get the actual child nodes.
+                        .filter(Objects::nonNull) // Filter out any null children.
```

Should we consider throwing an exception if a child is null?  Or potentially having this be a configurable thing (whether or not to throw an exception when a null child is encountered.)

---

I'm leaning towards making it configurable whether or not to throw an exception when a child is null. Do you have any preference for the default behavior?

---

I would throw an exception by default.  That way we will hopefully find mistakes like that during development and not when things are deployed in production.  In the event that throwing an exception is disabled, I would log an error.

---

Okay, I'll implement this behavior.